services/catalog/src/simcore_service_catalog/core/background_tasks.py

```diff
@@ -70,29 +70,36 @@ def _by_version(t: tuple[ServiceKey, ServiceVersion]) -> Version:
             (service_key, service_version)
         ]
         try:
-            ## Set deprecation date to null (is valid date value for postgres)
-
-            # DEFAULT policies
+            # 1. Evaluate DEFAULT ownership and access rights
             (
                 owner_gid,
                 service_access_rights,
-            ) = await access_rights.evaluate_default_policy(app, service_metadata)
+            ) = await access_rights.evaluate_default_service_ownership_and_rights(
+                app,
+                service=service_metadata,
+                product_name=app.state.default_product_name,
+            )
 
-            # AUTO-UPGRADE PATCH policy
-            inherited_access_rights = await access_rights.evaluate_auto_upgrade_policy(
-                service_metadata, services_repo
+            # 2. Inherit access rights from the latest compatible release
+            inherited_data = await access_rights.inherit_from_latest_compatible_release(
+                service_metadata=service_metadata,
+                services_repo=services_repo,
             )
 
-            service_access_rights += inherited_access_rights
+            # 3. Aggregates access rights and metadata updates
+            service_access_rights += inherited_data["access_rights"]
             service_access_rights = access_rights.reduce_access_rights(
                 service_access_rights
             )
 
-            # set the service in the DB
+            metadata_updates = {
+                **service_metadata.model_dump(exclude_unset=True),
+                **inherited_data["metadata_updates"],
+            }
+
+            # 4. Create or update the service in the database
```

Comment:
general comment:
- adding comments that exactly are written as such in the function names is not necessary from my point of view. that is exactly the idea of clean code, no need to repeat it since the names make sense.